datumaro/plugins/data_formats/coco/exporter.py

```diff
@@ -208,7 +208,10 @@ def find_instance_parts(self, group, img_width, img_height):
 
         anns = boxes + polygons + masks
         leader = anno_tools.find_group_leader(anns)
-        bbox = anno_tools.max_bbox(anns)
+        if len(boxes) > 0:
+            bbox = anno_tools.max_bbox(boxes)
+        else:
+            bbox = anno_tools.max_bbox(anns)
```

Comment:
could you explain why this should be?

Reply:
As you know, An item may have multiple annotations for both COCO and Datumaro.
In the case of COCO, segmentation and bbox are grouped into a single annotation.
Thus, e.g. of annotations for an item for COCO is like as the follow:
[
    {....., bbox: [0.0, 1.0, 2.0, 3.0], segmentation: [1035, 30, 395, ...], ...},
    {....., bbox: [1.0, 1.0, 3.0, 2.0], segmentation: [10, 30, 3435, ...], ...},
    ....
]
When we import these annotations to Datumaro, we treat them to 4 annotation include 2 bbox and 2 segmentation (LRE or polygon)
To mark that some of them come from a single annotation, we give them ids thus, Datumaro annotations will be like the follow:
[
   {id: 1, type: bbox, ...},
   {id: 1, type: polygon, ...},
   {id: 2, type: bbox, ...},
   {id: 2, type: lre, ...},
]
when we export them back to COCO type, we should merge the annotations if they have the same id.
The code above merges the annotations.

When we determine the merged single bbox, we aggregate the candidates (anns), calculate bbox of each annon, and pick the maximum bbox to the merged bbox.
Thus, if a bbox from the segmentation annotation is larger then bbox from the bbox annotation,
we prefer the bbox from the segmentation and ignore bbox from the bbox annotation.
Because of this, when we import COCO, and export COCO back, many of bbox annotations are changed now.
I think this is weird.

The code changes above are to give a higher priority to bbox annotation when we determine the merged bbox annotation.
With this changes, we will ignore bbox from the segmentation annotation if we have any bbox annotation for an item.